packages/dds/tree/src/test/utils.ts

```diff
@@ -426,38 +420,90 @@ export class TestTreeProviderLite {
-			this.containerRuntimes.add(containerRuntime);
+			this.containerRuntimeMap.set(tree.id, containerRuntime);
 			tree.connect({
 				deltaConnection: runtime.createDeltaConnection(),
 				objectStorage: new MockStorage(),
 			});
-			(tree as Mutable<SharedTreeWithConnectionStateSetter>).setConnected = (
-				connectionState: boolean,
-			) => {
-				containerRuntime.connected = connectionState;
-			};
 			t.push(tree);
 		}
 		this.trees = t;
 	}
 
-	public processMessages(count?: number): void {
-		// In TurnBased mode, flush the messages from all the runtimes before processing the messages.
-		// Note that this does not preserve the order in which the messages were sent. To do so, tests should
-		// flush messages from individual runtimes in the order they were created.
-		if (this.flushMode === FlushMode.TurnBased) {
-			this.containerRuntimes.forEach((containerRuntime) => {
+	/**
+	 * To be used in TurnBased mode where messages that are sent are not automatically flushed.
+	 * Flush the messages that have been sent by a given tree.
+	 * @param treeId - The id of the tree to flush messages for.
+	 * @remarks This will be a no-op if the tree's runtime is not connected.
+	 */
+	public flushMessages(tree: Pick<ISharedTree, "id">): void {
+		const containerRuntime = this.containerRuntimeMap.get(tree.id);
+		assert(containerRuntime !== undefined, "No tree found to flush messages");
+		containerRuntime.flush();
+	}
+
+	/**
+	 * Process the messages across all trees.
+	 * By default, this method also flushes the messages from all trees before processing them. In TurnBased mode,
+	 * messages are not automatically flushed so this ensures that all messages are processed. Flushing can be
+	 * disabled by setting the flush parameter to false. Then it is the responsibility of the test to flush the
+	 * messages before calling this method.
+	 * @param flush - Whether or not to flush the messages before processing them. Defaults to true.
+	 * @remarks
+	 * - Trees whose runtime is paused or not connected will queue these messages and not process them. They
+	 * will process the messages when their runtime is unpaused or connected respectively.
+	 * - Flushing does not preserve the order in which the messages were sent. To do so, tests should
+	 * call flushMessages on individual trees in the order messages were sent.
+	 */
+	public processMessages(flush: boolean = true): void {
+		if (flush) {
+			this.containerRuntimeMap.forEach((containerRuntime) => {
 				containerRuntime.flush();
 			});
 		}
-		this.runtimeFactory.processSomeMessages(
-			count ?? this.runtimeFactory.outstandingMessageCount,
-		);
+		this.runtimeFactory.processAllMessages();
+	}
+
+	/**
+	 * Process the given count of messages across all trees.
+	 * @remarks trees whose runtime is paused or not connected will queue these messages and not process them. They
+	 * will process the messages when their runtime is unpaused or connected respectively.
```

Comment:
Should this be "unpaused and connected"? I'm just thinking of the case where a tree (its runtime) is both paused and disconnected. When such a tree is re-connected, it still won't process the messages because its runtime will still be paused, right? (Same if the runtime is unpaused while the tree is still unconnected)

Reply:
Updated the comment.